packages/core/src/shared/sam/sync.ts

```diff
@@ -660,12 +592,13 @@ export async function prepareSyncParams(
     } else if (arg instanceof vscode.Uri) {
         if (arg.path.endsWith('samconfig.toml')) {
             // "Deploy" command was invoked on a samconfig.toml file.
+            // TODO: add step to verify samconfig content to skip param source prompter
             const config = await SamConfig.fromConfigFileUri(arg)
             const params = getSyncParamsFromConfig(config)
             const projectRoot = vscode.Uri.joinPath(config.location, '..')
             const templateUri = params.templatePath
                 ? vscode.Uri.file(path.resolve(projectRoot.fsPath, params.templatePath))
-                : undefined
+                : await getTemplateFile(projectRoot)
```

Comment:
Discuss with team to that we should look for the template file in the project root folder of there is no `template_path` or `templatePath` in customer `samconfig.toml` file

Reply:
Revert the change from current PR since with would require more refactoring. Will address this in separate PR.